pkg/tsdb/cloudwatch/request_parser_test.go

```diff
@@ -78,6 +82,99 @@ func TestRequestParser(t *testing.T) {
 				So(res.Dimensions["InstanceType"][0], ShouldEqual, "test2")
 				So(*res.Statistics[0], ShouldEqual, "Average")
 			})
+
+			Convey("period defined in the editor by the user is being used", func() {
+				query := simplejson.NewFromAny(map[string]interface{}{
+					"refId":      "ref1",
+					"region":     "us-east-1",
+					"namespace":  "ec2",
+					"metricName": "CPUUtilization",
+					"id":         "",
+					"expression": "",
+					"dimensions": map[string]interface{}{
+						"InstanceId":   "test",
+						"InstanceType": "test2",
+					},
+					"statistics": []interface{}{"Average"},
+					"hide":       false,
+				})
+				Convey("when time range is short", func() {
+					query.Set("period", "900")
+					timeRange := tsdb.NewTimeRange("now-1h", "now-2h")
+					from, _ := timeRange.ParseFrom()
+					to, _ := timeRange.ParseTo()
+
+					res, err := parseRequestQuery(query, "ref1", from, to)
+					So(err, ShouldBeNil)
+					So(res.Period, ShouldEqual, 900)
+				})
+			})
+
+			Convey("period is parsed correctly if not defined by user", func() {
+				query := simplejson.NewFromAny(map[string]interface{}{
+					"refId":      "ref1",
+					"region":     "us-east-1",
+					"namespace":  "ec2",
+					"metricName": "CPUUtilization",
+					"id":         "",
+					"expression": "",
+					"dimensions": map[string]interface{}{
+						"InstanceId":   "test",
+						"InstanceType": "test2",
+					},
+					"statistics": []interface{}{"Average"},
+					"hide":       false,
+					"period":     "auto",
+				})
+
+				Convey("when time range is short", func() {
+					query.Set("period", "auto")
+					timeRange := tsdb.NewTimeRange("now-2h", "now-1h")
+					from, _ := timeRange.ParseFrom()
+					to, _ := timeRange.ParseTo()
+
+					res, err := parseRequestQuery(query, "ref1", from, to)
+					So(err, ShouldBeNil)
+					So(res.Period, ShouldEqual, 60)
+				})
+
+				Convey("when time range is 5y", func() {
+					timeRange := tsdb.NewTimeRange("now-5y", "now")
+					from, _ := timeRange.ParseFrom()
+					to, _ := timeRange.ParseTo()
+
+					res, err := parseRequestQuery(query, "ref1", from, to)
+					So(err, ShouldBeNil)
+					So(res.Period, ShouldEqual, 21600)
+				})
+			})
+
+			Convey("closest works as expected", func() {
+				periods := []int{60, 300, 900, 3600, 21600}
+				Convey("and input is lower than 60", func() {
```

Comment:
@sunker Would you add test which confirm the period isn't shorter than CloudWatch retension?

https://aws.amazon.com/about-aws/whats-new/2016/11/cloudwatch-extends-metrics-retention-and-new-user-interface/

Sample test case.
```
So(closest(periods, (60 * 60 * 24 * 15) + 1), ShouldEqual, 300)
So(closest(periods, (60 * 60 * 24 * 63) + 1), ShouldEqual, 3600)
```